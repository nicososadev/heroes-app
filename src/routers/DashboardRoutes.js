import React, { Fragment } from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { DcPage } from '../components/dc/DcPage'
import { HeroPage } from '../components/heroes/HeroPage'
import { MarvelPage } from '../components/marvel/MarvelPage'
import { SearchPage } from '../components/search/SearchPage'
import { Navbar } from '../components/ui/NavBar'

export const DashboardRoutes = () => {
    return (
        <Fragment>
            <Navbar />

            <div className="container mt-3">
                <Switch>
                    <Route exact path='/marvel' component={ MarvelPage } />
                    <Route exact path='/dc' component={ DcPage } />
                    <Route exact path='/hero/:heroId' component={ HeroPage } />
                    <Route exact path='/search' component={ SearchPage } />

                    <Redirect to='/marvel' />
                </Switch>
            </div>
        </Fragment>
    )
}
