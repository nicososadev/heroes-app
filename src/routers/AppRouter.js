import React, { useContext } from 'react'
import { BrowserRouter as Router, Switch } from "react-router-dom";
import { AuthContext } from '../auth/AuthContext';
import { LoginPage } from '../components/login/LoginPage';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {

    const { user } = useContext(AuthContext)

    return (
        <Router>
            <div>
                <Switch>
                    <PublicRoute exact path='/login' isAuthenticated={ user.logged } component={ LoginPage } />
                    <PrivateRoute path='/' isAuthenticated={ user.logged } component={ DashboardRoutes } />
                </Switch>
            </div>
        </Router>
    )
}
