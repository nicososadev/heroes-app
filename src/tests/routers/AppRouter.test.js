import React from 'react'
import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { AuthContext } from '../../auth/AuthContext'
import { AppRouter } from '../../routers/AppRouter'

describe('Pruebas en el componente AppRouter', () => {

    const contextValue = {
        user: {
            logged:false
        },
        dispatch: jest.fn()
    }

    test('Debe mostrar el login si el usuario no esta autenticado ', () => {
        
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect( wrapper ).toMatchSnapshot()
    })

    test('Debe mostrar el componente de Marvel si el usuario esta autenticado', () => {
        
        const contextValue = {
            user: {
                name: 'Nicolas',
                logged:true
            },
            dispatch: jest.fn()
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        )

        expect( wrapper.find('HeroesList').exists() ).toBe( true )
    })
    

})
