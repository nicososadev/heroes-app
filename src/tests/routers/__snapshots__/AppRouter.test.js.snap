// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AppRouter Debe mostrar el login si el usuario no esta autenticado  1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
